x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/handle_api_errors.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpResponse } from 'src/core/public';
+
+import { FlashMessagesLogic, IFlashMessage } from './';
+
+/**
+ * The API errors we are handling can come from one of two ways:
+ *  - When our http calls recieve a response containing an error code, such as a 404 or 500
+ *  - Our own JS while handling a successful response
+ *
+ * In the first case, if it is a purposeful error (like a 404) we will receive an
+ * `errors` property in the response's data, which will contain messages we can
+ * display to the user.
+ */
+interface IErrorResponse {
+  statusCode: number;
+  error: string;
+  message: string;
+  attributes: {
+    errors: string[];
+  };
+}
+
+/**
+ * Converts API/HTTP errors into user-facing Flash Messages
+ */
+export const handleAPIError = (error: HttpResponse<IErrorResponse>) => {
```

Comment:
Also wanted to raise that I removed the [`onErrorCallback`](https://github.com/elastic/ent-search/blob/master/app/javascript/app_search/utils/handleAPIError.ts#L44) arg because I thought it was cleaner to have the helper be in charge of setting FlashMessages vs. passing in `setFlashMessages` each time.

It's possible we'll want/need to pass a custom callback in the future, but I can't think of a really useful case for a passed callback currently (esp with try/catch syntax) - can anyone else?

Another thought: since I'm making this handler specific to flash messages (I placed it in `shared/flash_messages` because I thought it made a lot of sense in there), should we consider renaming this to something like `flashAPIErrors` instead?

Reply:
I think this approach is fine since its coupled to the messages.

Reply:
Rad! I went ahead renamed `handleAPIError` to `flashAPIErrors` here: ad4f8fd

CC @byronhulcher as a heads up!